docs/architecture/mcu-protocol.md

```
@@ -0,0 +1,268 @@
+    mutex_unlock(&dev->mcu.mutex);
+    return ret;
+}
+```
+
+## Error Handling
+
+### Timeout Handling
+
+If MCU doesn't respond within timeout:
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)